vowpalwabbit/array_parameters.h

```diff
@@ -0,0 +1,172 @@
+#pragma once
+#include <stdio.h>
+#include <iostream>
+#include <stdlib.h>
+#include <string.h>
```

Comment:
I don't think any of these headers are really needed, with the possible exception of string.h (for memcpy below).  Having said that, the correct header names for C++ are actually <cstdio>, <stdlib> and <cstring>.